### PR TITLE
Hotfix story load creatures

### PR DESCRIPTION
**Pull request by 6fears7**
Was investigating missing critters:

```
[Info   :RainMeadow] 18:31:43|914365|SpectatorOverlay.Update:True
[Info   :RainMeadow] 18:31:43|914365|RPCManager.BuildRPC:Sending RPC: Void GoToDeathScreen()

```

we're entering the next process still considered spectating 

- [x] Test
Tested, works.

### CODE REVIEW

File: Story/OnlineUIComponents/SpectatorOverlay.cs [hunk 1]
```diff
@@ -49,8 +49,10 @@ public PlayerButton(SpectatorOverlay menu, OnlinePhysicalObject opo, Vector2 pos
                 this.button.OnClick += (_) =>
                 {
                     this.button.toggled = !this.button.toggled;
-                    overlay.spectatee = this.button.toggled ? opo.apo as AbstractCreature : null;
-                    OnlineManager.mePlayer.isActuallySpectating = overlay.spectatee == null || !this.player.isMine;
+                    overlay.spectatee = this.button.toggled ? this.player.apo as AbstractCreature : null;
+                    bool result = !this.player.isMine ? true : overlay.spectatee == null ? false : false;
```
Comment (6fears7) [inline Story/OnlineUIComponents/SpectatorOverlay.cs:53]: Maybe we could just simplify to 

OnlineManager.mePlayer.isActuallySpectating = !this.player.isMine? What's the thinking behind == null?

File: Game/RainMeadow.LoadingHooks.cs [hunk 2]
```diff
@@ -226,6 +226,7 @@ private void WorldLoader_ctor(On.WorldLoader.orig_ctor_RainWorldGame_Name_bool_s
         {
             if (OnlineManager.lobby != null)
             {
+                OnlineManager.mePlayer.isActuallySpectating = false;
```
Comment (6fears7) [inline Game/RainMeadow.LoadingHooks.cs:229]: I think it's best here so we cover death / karma, or its usage in arena. lmk what you think.

Comment (6fears7) [reply Game/RainMeadow.LoadingHooks.cs:229]: @youbitchoc pinging you for awareness prior to merge.

Comment (youbitchoc) [reply Game/RainMeadow.LoadingHooks.cs:229]: we set session init books elsewhere, iinw it's in getorinitiatesavestate

File: Story/OnlineUIComponents/SpectatorOverlay.cs [hunk 3]
```diff
@@ -49,7 +49,7 @@ public PlayerButton(SpectatorOverlay menu, OnlinePhysicalObject opo, Vector2 pos
                 this.button.OnClick += (_) =>
                 {
                     this.button.toggled = !this.button.toggled;
-                    overlay.spectatee = this.button.toggled ? opo.apo as AbstractCreature : null;
+                    overlay.spectatee = this.button.toggled ? this.player.apo as AbstractCreature : null;
```
Comment (youbitchoc) [inline Story/OnlineUIComponents/SpectatorOverlay.cs:52]: why the change?

**Review by youbitchoc — CHANGES_REQUESTED** [review]

File: Game/RainMeadow.LoadingHooks.cs (hunk 2 above)
Comment (6fears7) [reply Game/RainMeadow.LoadingHooks.cs:229]: I'll look and see. I personally think we should set this as high up the stack as possible to prevent any unwanted problems arising

File: Story/OnlineUIComponents/SpectatorOverlay.cs (hunk 3 above)
Comment (youbitchoc) [reply Story/OnlineUIComponents/SpectatorOverlay.cs:52]: sure ig

File: Game/RainMeadow.LoadingHooks.cs (hunk 2 above)
Comment (youbitchoc) [reply Game/RainMeadow.LoadingHooks.cs:229]: we can also shove it in StoryGameSession_ctor

Comment (6fears7) [reply Game/RainMeadow.LoadingHooks.cs:229]: My only concern with storyGameSess is if other game modes want to use it at some point.

**Comment by youbitchoc** [comment]
lgtm
